src/lib/util/logger/Logger.ts

```diff
@@ -0,0 +1,229 @@
+	): this;
+
+	once<K extends keyof GClientEvents>(event: K, listener: (...args: GClientEvents[K]) => Awaitable<void>): this;
+	once<S extends string | symbol>(
+		event: Exclude<S, keyof GClientEvents>,
+		listener: (...args: any[]) => Awaitable<void>,
+	): this;
+
+	emit<K extends keyof GClientEvents>(event: K, ...args: GClientEvents[K]): boolean;
+	emit<S extends string | symbol>(event: Exclude<S, keyof GClientEvents>, ...args: unknown[]): boolean;
+
+	off<K extends keyof GClientEvents>(event: K, listener: (...args: GClientEvents[K]) => Awaitable<void>): this;
+	off<S extends string | symbol>(
+		event: Exclude<S, keyof GClientEvents>,
+		listener: (...args: any[]) => Awaitable<void>,
+	): this;
+
+	removeAllListeners<K extends keyof GClientEvents>(event?: K): this;
+	removeAllListeners<S extends string | symbol>(event?: Exclude<S, keyof GClientEvents>): this;
+}
+
+declare module 'discord.js' {
+	// eslint-disable-next-line @typescript-eslint/no-empty-interface
+	interface ClientEvents extends GClientEvents {}
+}
+
+export const enum LogLevel {
+	TRACE = 1,
+	DEBUG = 2,
+	INFO = 3,
+	/**
+	 * @deprecated
+	 */
+	TIME = 4,
+	WARN = 5,
+	ERROR = 8,
+	OFF = 99,
+}
+
+export type LogMethods = 'trace' | 'debug' | 'info' | 'time' | 'warn' | 'timeEnd' | 'error';
+
+export class ILogger {
+	TRACE: LogLevel.TRACE;
+	DEBUG: LogLevel.DEBUG;
+	INFO: LogLevel.INFO;
+	TIME: LogLevel.TIME;
+	WARN: LogLevel.WARN;
+	ERROR: LogLevel.ERROR;
+	OFF: LogLevel.OFF;
+	level: LogLevel = LogLevel.TRACE;
+
+	constructor() {
+		this.TRACE = LogLevel.TRACE;
+		this.DEBUG = LogLevel.DEBUG;
+		this.INFO = LogLevel.INFO;
+		this.TIME = LogLevel.TIME;
+		this.WARN = LogLevel.WARN;
+		this.ERROR = LogLevel.ERROR;
+		this.OFF = LogLevel.OFF;
+	}
+
+	public trace(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.TRACE, ...values);
+	}
+
+	public debug(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.DEBUG, ...values);
+	}
+
+	public info(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.INFO, ...values);
+	}
+
+	/**
+	 * @deprecated
+	 */
+	public time(val: string): void {
+		if (val.length > 0) this.invokeTime(LogLevel.TIME, val, 'start');
+	}
+
+	/**
+	 * @deprecated
+	 */
+	public timeEnd(val: string): void {
+		if (val.length > 0) this.invokeTime(LogLevel.TIME, val, 'stop');
+	}
+
+	public warn(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.WARN, ...values);
+	}
+
+	public error(...values: readonly unknown[]): void {
+		this.invoke(LogLevel.ERROR, ...values);
+	}
+
+	public invoke(level: LogLevel, ...values: readonly unknown[]): void {
+		if (!this.enabledFor(level)) return;
+
+		let color = '';
+		if (level === LogLevel.TRACE) color = '\x1b[91m';
+		else if (level === LogLevel.DEBUG) color = '\x1b[2m';
+		else if (level === LogLevel.INFO) color = '\x1b[36m';
+		else if (level === LogLevel.WARN) color = '\x1b[93m';
+		else color = '\x1b[91m';
+
+		const method = this.LevelMethods.get(level);
```

Comment:
```suggestion
		const method = this.LevelMethods.get(level) as 'error' | 'warn' | 'etc';
```
Maybe this fixes the ts error?